.github/workflows/publish-release.yml

```diff
@@ -0,0 +1,190 @@
+name: Publish Release
+
+on:
+  release:
+    types: [published]
+jobs:
+  Get-Properties:
+    runs-on: ubuntu-latest
+    outputs:
+      release-type: ${{ steps.type.outputs.release-type }}
+      carpet-version: ${{ steps.properties.outputs.mod_version }}
+      minecraft-version: ${{ steps.properties.outputs.minecraft_version }}
+      # extra-branch: ${{ steps.properties.outputs.release-extra-branch }} Not needed anymore
+      matrix-exclude-branch: ${{ steps.processmatrix.outputs.matrix-to-exclude }}
```

Comment:
what is this?

Reply:
At the beginning, I had a copy of all steps in order to build two branches, since I didn't know you could exclude matrix things dynamically.

Matrix jobs allow you to run a job multiple times with different "properties", and they have an `exclude` parameter that is normally used to exclude combinations (eg don't run Java 16 on Windows), but in this case, if snapshots aren't configured to run, this will output `Snapshots`, so the `Snapshots` job in the matrix will be completely excluded.

That way, all the code is not duplicated and I don't have to make it fail to work (previously I literally had `if !buildextra failhard()` for the snapshots branch, this looks and works A LOT better).

Reply:
ah, i just got confused with #656